implementations/scripts/Metric.py

```diff
@@ -0,0 +1,41 @@
+import pandas as pd
+
+
+class Metric():
+    """
+    Create a dataframe from data-list, which is the data
+    fetched by Perceval
+
+    :param data_list: A list of dictionaries.
+        Each element is a Perceval dictionary, obtained from a JSON
+        file or from Perceval directly.
+    """
+
+    def __init__(self, data_list):
+        data_list_flattened = Metric._flatten_data(data_list)
+        self.raw_df = pd.DataFrame(data_list_flattened)
+
+    @staticmethod
+    def _flatten_data(data_list):
```

Comment:
I'm not sure we're on the same page with respect to this method. As i commented above, I think this should be a function provided by children classes. For example, `Commit` (or maybe `Code_Changes`) will define it to get the fields from the Perceval dictionary that you really need to produce the flat directory that you need to produce a row in the dataframe. You don't really need all fields in the original Perceval directory, only some of them. `Commit` or maybe `Code_Changes` would know which fields are actually needed for implementing the metric and its filters.

Reply:
I agree with what you say. But if `_flatten_data` is defined in `Commit`, where it will directly act on Perceval data and flatten it while keeping only those fields which are required, what is the need for the root Metric class? This is where I am stuck.
Do you want it so that the Metric(root) class is never explicitly called and just acts like a template of each of its children (the commit, issue and pull request classes)?
The way I am using these classes as of now is to create a object for every individual metric class (like CodeChanges, for example), and each class calls its parent's `__init__`

Please let me know. 
As of now, we have three main "checks":
- date range  
- source code  
- keeping only required fields.

**date range** is common to each and every metric we implement, irrespective of category (commit, issue, pull request).
**source code**: as of now, I can't think of a way to check for issues (but we will see that later)
**keeping only required fields**: This is specific to each Category.